### PR TITLE
qt5 5.5.0 

### PR DESCRIPTION
**Pull request by DomT4**
While the bot is quiet I wanted to test whether the `rpath` changes were causing the build failures on the test-bot, and integrate Secure Transport/No OpenSSL. Commit history from #41252 preserved.

### CODE REVIEW

File: Library/Formula/qt5.rb [hunk 1]
```diff
@@ -43,20 +51,15 @@ class Qt5 < Formula
...
   depends_on OracleHomeVarRequirement if build.with? "oci"
 
   def install
     ENV.universal_binary if build.universal?
 
     args = ["-prefix", prefix,
-            "-system-zlib",
+            "-system-zlib", "-securetransport",
             "-qt-libpng", "-qt-libjpeg",
+            "-no-rpath", "-no-openssl",
```
Comment (mistydemeo) [inline Library/Formula/qt5.rb]: Are we breaking the rpath install names via our install name rewriting? Is that something we can fix?

Comment (DomT4) [inline Library/Formula/qt5.rb]: I'm not completely sure what's going on here. `rpath` [apparently became](https://github.com/Homebrew/homebrew/pull/41252#issuecomment-118557153) the default for Qt5 in this release, and the bot errors imply (Although perhaps not correctly) that the `rpath` setting wasn't linking as expected for some reason.

I did a full build locally running with the same `--build-bottle` variables as the test-bot and both the installation and bottle turned out fine, which makes me wonder if I'm getting lucky or if something about the test-bot is borking the installation in some way. We disable rpath usage in a few formulae, but it isn't widespread. There are build logs from my local install [here](https://gist.github.com/anonymous/d2047f4c8db4e58adc15) if that's any use at all.

Comment (DomT4) [inline Library/Formula/qt5.rb]: [This is everything we're changing the paths on after installation](https://gist.github.com/DomT4/0f962ab0e55264c0f37f).

**Comment by jakepetroules** [comment]
> Are we breaking the rpath install names via our install name rewriting? Is that something we can fix?

@mistydemeo Is this something Homebrew does automatically for all binaries of installed formulae?

**Comment by DomT4** [comment]
Dropping the `rpath` change has allowed everything but Qt's test to pass on Mountain Lion, so perhaps `rpath` usage is the problem:

```
==> brew test --verbose qt5						 FAILED
Testing qt5
==> /usr/local/Cellar/qt5/5.5.0/bin/qmake -project
Error: qt5: failed
execution expired
/usr/local/Library/Homebrew/formula.rb:868:in `gets'
/usr/local/Library/Homebrew/formula.rb:868:in `block in system'
/usr/local/Library/Homebrew/formula.rb:854:in `open'
/usr/local/Library/Homebrew/formula.rb:854:in `system'
/usr/local/Library/Formula/qt5.rb:130:in `block in <class:Qt5>'
/usr/local/Library/Homebrew/formula.rb:797:in `block in run_test'
/usr/local/Library/Homebrew/extend/fileutils.rb:17:in `mktemp'
/usr/local/Library/Homebrew/formula.rb:793:in `run_test'
/usr/local/Library/Homebrew/test.rb:28:in `block in <main>'
/usr/local/Library/Homebrew/test.rb:27:in `<main>'
==> brew fetch --retry pyqt5						 PASSED
==> brew install pyqt5							 PASSED
==> brew test --verbose pyqt5						 PASSED
==> brew fetch --retry qbs						 PASSED
==> brew install qbs							 PASSED
==> brew test --verbose qbs						 PASSED
```

**Comment by DomT4** [comment]
Slightly different error. Our bottles are now doing this:

```
qt5/5.5.0/bin/qdoc
qt5/5.5.0/bin/qhelpconverter
qt5/5.5.0/bin/qhelpgenerator
qt5/5.5.0/bin/qlalr
qt5/5.5.0/bin/qmake
gzip: (stdin): unexpected end of file
gtar: Unexpected EOF in archive
gtar: Unexpected EOF in archive
gtar: Error is not recoverable: exiting now
```

**Comment by luzpaz** [comment]
Thanks for your work on this so far

**Comment by luzpaz** [comment]
checking in to see if there is anything that I can do to help this get merged ?

**Comment by DomT4** [comment]
I want to do another local build to see if `qmake` does actually work, the installation worked last time but I didn't test it properly, and another full local build outside of Homebrew to see if it is our `install_name_tool` changes causing the breakages.

The issue has just been finding ~6 hours of time I don't need to do anything else on the machine for, since Qt essentially sets fire to the CPU during compile :smile_cat:

**Comment by DomT4** [comment]
Alright, from-source outside-of-brew build has been started. Shall see if the end product is usable, and then do the exact same build inside `brew` and see if that is usable, and that should help determine whether our `install_name` changes are the culprit here.

**Comment by DomT4** [comment]
Either the `qmake` bug exists in the from-source, outside-brew build or `qmake -project` now hangs unless there's actually a project. The latter is more likely and would explain the timeout on the `qt5` test itself at least.

Was able to build a `qmake`-using project just fine with the from-source, outside-brew Qt5 build. No issues building or running a project built against it.

Will set an inside-brew build running in a few hours and see if the same functionality exists.

**Comment by luzpaz** [comment]
how's this going so far ?

**Comment by luzpaz** [comment]
@DomT4 how did the inside brew build go ?

**Comment by luzpaz** [comment]
Sorry I know this is annoying but.. 
bump
:disappointed:

**Comment by DomT4** [comment]
Sorry. `qt5` is an absolute killer to build, and I essentially can't touch the laptop whilst it does the 3+ hour build without burning a hole in my lap (105+ degree CPU temps, woo), which is difficult when I have a lot of uni work to do. I'll try and tee it up when I go sleep later and let it build then.

**Comment by luzpaz** [comment]
@DomT4 I know how that feels like since I've built it before and it's an atrocious amount of time. Thanks so much for your work on this so far!

**Comment by mikemcquaid** [comment]
Maybe it's worth trying this without secure transport for now?

**Comment by DomT4** [comment]
Secure Transport doesn't seem to be the breaking variable. It broke before we checked out the Secure Transport option as well. It seems to be `rpath` related, or something we do with our `install_names` after installation, but proving fun to narrow down.

**Comment by luzpaz** [comment]
@DomT4 Can you post gist-logs so we can use the power of the crowd to solve this ?

**Comment by DomT4** [comment]
`gist-logs` are in the earlier diff with Misty. The important `install_name` changes aren't added automatically to those logs, but I put a link to them there as well.

Currently doing a fresh install. Should be done in about ~9 years/4 hours.

**Comment by DomT4** [comment]
New gist-logs [here](https://gist.github.com/anonymous/b987f06d298e299a088c).

`install_name` changes [here](https://gist.github.com/DomT4/016a4ce9ceeee2b45361). The latter is a 2500 line document :whale2:.

**Comment by DomT4** [comment]
`qbs` will build fine against this `qt5` as well. It could be that `qbs` doesn't exercise every last Framework Qt ships, so I don't want us reading too hugely into the will it/won't it compile, but it is a useful baseline.

Am going to fix up Qt5's test, and then set the bot running on this again, and cross all of my fingers.

**Comment by DomT4** [comment]
Alright, build is a *go*. Have fixed the test so it doesn't hang and die horribly. Now I just cross my fingers for the next 4 hours whilst the test-bot does its thing and hope the bottle that pops out of the end is usable this time.

**Comment by mikemcquaid** [comment]
Looks like it built. Hooray.

**Comment by mikemcquaid** [comment]
Merged this to avoid any :ghost:s ruining things.

**Comment by DomT4** [comment]
The bottle will untar this time as well. I don't think I've ever been happier to get a PR passing and behaving as expected. This was a nightmare :sweat_smile:.

**Comment by luzpaz** [comment]
What a trip! Thanks @DomT4 
You Rock! :+1:

**Pull request closed without merging**